framework/src/geomsearch/FindContactPoint.C

```diff
@@ -171,32 +175,75 @@ findContactPoint(PenetrationInfo & p_info,
     }
 
     DenseVector<Real> rhs(dim - 1);
-
     rhs(0) = -dxyz_dxi[0] * d;
 
     if (dim - 1 == 2)
       rhs(1) = -dxyz_deta[0] * d;
 
     DenseVector<Real> update(dim - 1);
-
     jac.lu_solve(rhs, update);
 
-    ref_point(0) += update(0);
+    // Improvised line search in case the update is too large and gets out of the element so bad
+    // that we cannot reinit at the new point
+    Real mult = 1;
+    while (true)
+    {
+      try
+      {
+        ref_point(0) += mult * update(0);
 
-    if (dim - 1 == 2)
-      ref_point(1) += update(1);
+        if (dim - 1 == 2)
+          ref_point(1) += mult * update(1);
 
-    points[0] = ref_point;
-    fe_side->reinit(side, &points);
-    d = secondary_point - phys_point[0];
+        points[0] = ref_point;
+        fe_side->reinit(side, &points);
+        d = secondary_point - phys_point[0];
 
-    update_size = update.l2_norm();
+        // we don't multiply by 'mult' because it is used for convergence
+        update_size = update.l2_norm();
+        break;
+      }
+      catch (libMesh::LogicError & e)
+      {
+        ref_point(0) -= mult * update(0);
+        if (dim - 1 == 2)
+          ref_point(1) -= mult * update(1);
+
+        mult *= 0.5;
+        if (mult < 1e-6)
+        {
+#ifndef NDEBUG
+          mooseWarning("We could not solve for the contact point.", e.what());
```

Comment:
using the macro in lieu of 'verbose' since I would have to add verbose booleans to 3 classes to propagate the option